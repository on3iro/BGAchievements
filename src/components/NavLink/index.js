/**
  * Renders a single NavLink item containing an li and a preformatted react-router Link
  *
  * @namespace Navigation.NavLink
  * @memberOf Navigation
  */

import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { NavLink } from 'react-router-dom'

import { surpressNoUsedVarsLinterError } from 'shared/helperFunctions'

export const Li = styled.li`
  display: inline-block;
  height: 100%;

  text-align: center;

  &:hover a {
    color: ${props => props.theme.main};
  }

  @media (max-width: 900px) {
    display: block;
  }
`

export const RestyledLink = styled(NavLink)`
  display: inline-block;
  height: 100%;
  padding: 5px 25px;

  color: ${props => props.theme.noneWhite};
  font-size: 1.4rem;
  font-family: ${props => props.theme.primaryFont};
  text-decoration: none;

  outline: 0;

  &.active {
    color: ${props => props.theme.main};
  }
`

const StyledNavLink = props => {
  // pull props off to avoid react-unknown-prop error
  const { icon, iconHover, secondary, ...rest } = props
  surpressNoUsedVarsLinterError({ icon, iconHover, secondary })

  return (
    <Li>
      <RestyledLink
        {...rest}
      >
        {props.children}
      </RestyledLink>
    </Li>
  )
}

StyledNavLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  icon: PropTypes.string,
  iconHover: PropTypes.string,
  secondary: PropTypes.bool
}

export default StyledNavLink
