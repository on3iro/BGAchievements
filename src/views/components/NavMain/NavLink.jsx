// @flow

import React from 'react'
import styled from 'styled-components'
import { NavLink } from 'react-router-dom'

const LinkWithActiveClass = props => (
  <NavLink {...props} activeClassName='selected' />
)

const StyledNavLink = styled(LinkWithActiveClass)`
  display: block;
  padding: 10px;

  color: ${props => props.theme.white};

  cursor: pointer;
  text-decoration: none;

  &.selected {
    background: ${props => props.theme.bgPrimary};
    color: ${props => props.theme.primary};

    &:visited {
      color: ${props => props.theme.primary};
    }
  }

  &:visited {
    color: inherit;
  }
`

export {
  StyledNavLink as default
}
