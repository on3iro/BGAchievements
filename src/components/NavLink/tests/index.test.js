import React from 'react'
import { shallow } from 'enzyme'

import NavLink, { Li, RestyledLink } from '../index'

describe('<NavLink />', () => {
  it('should render styled.li', () => {
    const renderedWrapper = shallow(<NavLink to='/'>PasswordGenerator</NavLink>)
    expect(renderedWrapper.find(Li).length).toMatchSnapshot()
  })

  it('should render <RestyledLink />', () => {
    const renderedWrapper = shallow(<NavLink to='/'>PasswordGenerator</NavLink>)
    expect(renderedWrapper.find(RestyledLink).length).toMatchSnapshot()
  })
})
