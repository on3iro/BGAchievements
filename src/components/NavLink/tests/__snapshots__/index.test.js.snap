exports[`<NavLink /> should render <RestyledLink /> 1`] = `1`

exports[`<NavLink /> should render styled.li 1`] = `1`
